Umee-Bots/Large-Amount-Account-Involvement/src/agent.ts

```diff
@@ -0,0 +1,45 @@
+import { Finding, HandleTransaction, TransactionEvent, FindingSeverity, FindingType } from "forta-agent";
+import { CONFIG, AgentConfig } from "./agent.config";
+
+export const provideHandleTransaction = (config: AgentConfig): HandleTransaction => {
+  const monitoredAddresses = config.monitoredAddresses.map((el) => el.toLowerCase());
+
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+    const involvedAddresses: string[] = [];
+    const amountOfInvolvedAddresses: number = Object.keys(txEvent.addresses).length;
+
+    monitoredAddresses.forEach((address) => {
+      if (address in txEvent.addresses) {
+        if (amountOfInvolvedAddresses >= config.threshold) {
+          involvedAddresses.push(address);
+        }
+      }
+    });
+
+    if (involvedAddresses.length) {
+      findings.push(
+        Finding.from({
+          alertId: "UMEE-13",
+          name: "Large amount of account involvement",
+          description: "Transaction includes large amount of addresses",
+          type: FindingType.Info,
+          severity: FindingSeverity.Info,
+          protocol: "Umee",
+          metadata: {
+            from: txEvent.from,
+            to: txEvent.to || "",
+            monitoredAddresses: JSON.stringify(involvedAddresses),
```

Comment:
you can use the new addresses field, instead of using adding the addresses here as metadata. Please do it.

Reply:
done. thanks 👍